contract-examples/contracts/ExampleDeployerList.sol

```diff
@@ -4,12 +4,145 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "./IAllowList.sol";
 import "./AllowList.sol";
+import "ds-test/src/test.sol";
+
+address constant DEPLOYER_LIST = 0x0200000000000000000000000000000000000000;
+address constant OTHER_ADDRESS = 0x0Fa8EA536Be85F32724D57A37758761B86416123;
```

Comment:
Why not?

Reply:
These contracts were meant to be deployable/example contracts for a simple understanding how precompiles can be called from contracts. I think they become a bit hard to follow in this form with tests. We put these examples to provide a simple example for those developing with precompiles. and also add another layer of testing to precompiles. Adding any test logic to actual contracts might be confusing and could lead others to include those in their own contracts.

Reply:
Makes sense. It seems like it's also a little confusing that we have both example contracts and tests in the same place (all under `example-contracts`). Does it make sense to separate the two? It seems like based on original intent, `contract-examples` should be separated to its own repo, especially with potential vulnerabilities in npm modules (one of the original motivations for moving away from HardHat). 

@ceyonur, I think you already had some context here, but could use @aaronbuchwald's opinion as well (regarding splitting out the examples into its own repo). 

I believe we can trigger a set of tests in a new repo on every merge to `master` on this repo too. 

We would want to be sure that examples work (nothing as frustrating as examples that don't), but that should never block code from getting merged into this repo. We would then have a nice clean separation between what is an example and what is an e2e test.

Reply:
I agree. If we cannot get rid of NPM (or any insecure dep management) currently we should move this out of the repo. IMO we can even keep the old hardhat tests in this folder along with these new solidity tests. This can still helpful for developers to use precompiles with JS/TS functions.

We can move necessary files for e2e (under tests/e2e/precompile) to this new repo and run those tests with the latest subnet-evm (master?). I think we can even put contract-examples repo as a submodule under subnet-evm repo. I think if we use this as a submodule we can run those e2e solidity tests in PR branches as well(with a script replaces subnet-evm via `go replace github.com/ava-labs/subnet-evm => ../`). This can be run as part of subnet-evm e2e tests and can report pass/failure before merging the PR to master.

Reply:
What I'm suggesting a copy-paste of `contract-examples` from `master` (before any of my changes, so, excluding `utils.ts`). It should still use `HardHat` as that seems to be a community best practice for integration tests (`Forge` is more of a unit-testing framework since you're stubbing out the EVM implementation for `hevm`) 

Then this repo, we can actually use `Foundry` to compile all the solidity and rely on `ethers` and `mocha` as the only dependencies. `npm` itself isn't inherently flawed from a security standpoint, it's just the dependencies. `ethers` doesn't have sub-dependencies and `mocha` has very few, but they are *very* well vetted. 

@ceyonur, what do you think? I can actually spin up that repo and make sure the tests work before we merge anything here. I think it makes sense to have working examples somewhere before these changes are merged. A little scope creep, but I think it's for the better.

cc @aaronbuchwald

Reply:
Also agree the contracts functions should be simple and just call the precompile. People fork subnet-evm all the time and stateful precompiles need to be as simple as possible.  Also SGTM on moving the example contracts to another repo just for us to hardcore test.

Reply:
As long as we can keep running e2e tests through github actions on PRs to this repository, I don't really care where they are placed (maybe git submodules).

Reply:
SGTM ✅ 

I think we can also copy e2e tests to that new example repository so we can run those hardhat tests in that repository without manually running the subnet-evm network elsewhere. 

Optionally we can add that repository as a submodule to subnet-evm repo so it would be easier to maintain the contract-examples repo when we add a new precompile.

Reply:
From: https://github.com/ava-labs/subnet-evm/pull/601#discussion_r1181834836

I think these are 2 different issues. I still strongly think we should separate out test logic from actual contracts (regardless of moving hardhat tests into new repo). I don't think this is related to Foundry/Hardhat choice directly but to keeping things minimal and readable in general. 

Do we have any concern about using a separate test file for contracts?

Reply:
Hmm if I did not understand this wrong we're about to remove those examples from this repository and turn them into test contracts that will contain no extra logic (like erc20/c-chain&wagmi fees etc)? Previously I was under impression that we will keep those examples in both repositories.

The plan sounds good, we can simplify those contracts and turn them into raw test contracts that will test only precompile logic. However I don't still think it's an utter necessity to keep them together until that point; especially if we want to merge this one to master without waiting the other.

For some clarification: In a perfect world we should not need those solidity tests for precompile testing. We already have unit tests in their related packages, and integration tests under `plugin/evm/vm_test.go`. However our stateful precompiles are a bit different than EVM precompiles. We wanted to make sure we're not breaking any invariant there and put up some wrapper contracts which also can be used as examples. Instead of testing the precompile logic directly (which should be tested by go tests), we tested the example contracts and related precompiles through them. In the process we might have added some extra assertions for precompiles directly to make sure our test paths are correctly executed.

Reply:
>However I don't still think it's an utter necessity to keep them together until that point

It's more work to move them to separate files, import all the examples, then when we go to delete the examples, we would want to move all of the test contracts up a directory because `contracts` wouldn't contain anything besides `tests`. I guess it would also contain the interfaces, but I think it's fine for those to live together the same way we didn't have a separate directory for examples.

>For some clarification:...

Yep, makes sense. but now that we're writing our tests in Solidity, we are already testing calling the precompiles from a contract.